### ISSUE TITLE
Reproduction Question: Clarification on MMLU Sub-Categories in Table 1

### ISSUE THREAD

**Issue by Liam-L2** [Fighoture/MoE_unsupervised_pruning#2]
Hi, thank you for your excellent work! I found your paper very insightful, and Table 1 in particular demonstrates strong experimental results.

I had a quick question regarding the evaluation setup for the MMLU benchmark. In your results, MMLU is broken down into three categories: Computer Science, Math, and Business. However, I wasn't able to find these exact groupings in the official [lm-evaluation-harness](https://github.com/EleutherAI/lm-evaluation-harness) repository.

Did you use the lm-evaluation-harness tool for these experiments? And if so, could you kindly clarify how these three MMLU sub-categories were constructed? For example, were they formed by grouping certain subjects from the full MMLU suite?

Thanks in advance for your clarification, and again, great work!

**Comment by Fighoture** [comment]
Hello, thanks for your interest in our work. Here is the mapping rules:

mmlu_subject_to_pruned_mask = {"college_mathematics": "math",
                               "college_computer_science": "code",
                               "high_school_mathematics": "math",
                               "high_school_computer_science": "code",
                               "high_school_microeconomics": "business",
                               "high_school_macroeconomics": "business",
                               "econometrics": "business"}

We used these 7 categories in MMLU as benchmark for our referring 3 fields. Hope that will work.

**Comment by Liam-L2** [comment]
Thank you for your clarification regarding the MMLU category mapping.

I’ve recently attempted to reproduce your results using the Qwen1.5-MoE-A2.7Bbase model, evaluating it on the same 7 MMLU subjects you mentioned (college_mathematics, high_school_mathematics, college_computer_science, high_school_computer_science, high_school_microeconomics, high_school_macroeconomics, and econometrics). After averaging the scores within each group, I found that the Business category result (45.82) matched exactly with Table 1 from your paper. However, the scores for Math and Computer Science showed notable discrepancies from the reported values.

Could you kindly share any additional details about your evaluation setup? For example, did you use specific prompts, hyperparameters, or inference settings when running lm-eval-harness? I’m trying to understand what might have caused the deviations.

Thank you again for your time and excellent work. I greatly appreciate your insights!

**Comment by Fighoture** [comment]
Hello, you can refer my another repo called MMLU_series_benchmark, the script evaluate_from_local.py may help you. Thanks

**Comment by Liam-L2** [comment]
<img width="920" alt="Image" src="https://github.com/user-attachments/assets/0083df75-04f4-4a3b-ad11-55f0d9d57667" />
Thank you very much for sharing the repository. I now have a clear understanding of the testing procedure. However, I still have one question: are the results presented in the figures averaged across multiple datasets? If so, could you kindly provide an example—for instance, how the results are obtained for the Math dataset?

**Comment by Fighoture** [comment]
The first three columns is the average accuracy according to mmlu_subject_to_pruned_mask I introduced before. For example, Math column refers to the average accuracy of high_school_mathematics and college_mathematics. Average column is the average of the first three columns.

**Comment by Liam-L2** [comment]
First off, big thanks for always taking the time to reply in detail! Your explanations have been super helpful for me to grasp the testing process step by step.But here’s the snag I’m hitting. When I tested the unpruned Qwen - Moe (the baseline model) on GSM8K with the lm - evaluation - harness tool, the results just don’t match what’s in the table. And it’s not just GSM8K—I’ve run evaluations on other test sets using the same tool, yet none of them can reproduce the data shown.Also, the MMLU_series_benchmark you shared before focuses on pruned models. There’s no info on how the baseline model performs in those tests. Could you shed some light on how the baseline data was obtained? Maybe there are specific configurations, dataset versions, or extra steps for the baseline that I’m missing? Any guidance to help me align my results would be really appreciated!

**Comment by Fighoture** [comment]
I tested GSM8K on GSM8K-eval repo, and Humaneval on evalplus repo. Both of that could be found in my main page. You could reproduce from this two repos.

For MMLU_series_benchmark repo, you can also test any unpruned model by giving pruned_result configuration with an ".". This "." configuration also works for the repos I mentioned above for base model benchmarking. And the MMLU data could be obtained from the link in the readme file of that repo.

**Comment by Liam-L2** [comment]
Got it, I see where the problem was! I’d been stuck in the lm - evaluation - harness way of testing, which turned out to be different from yours. That’s why the results didn’t match. 

Big thanks for your super patient replies—you’re truly the most helpful and patient person I’ve met in my research journey. Wishing you all the best and a great day!

**Comment by Liam-L2** [comment]
Hello,I have followed your suggested testing method and conducted tests on the Business task using both the Qwen and DeepSeek baseline models (unpruned). However, the results are still inconsistent.

Please find below my test results and the corresponding code for your reference.
I would appreciate it if you could help analyze the cause of the issue. Looking forward to your response.

qwen,CoT,high_school_microeconomics-high_school_macroeconomics-econometrics,06-20_19-49,0.4514824791758996
deepseek,CoT,high_school_microeconomics-high_school_macroeconomics-econometrics,06-20_20-35,0.5350404305457677

<img width="1109" alt="Image" src="https://github.com/user-attachments/assets/a318d10b-a607-4c16-9827-838a7aca8396" />

<img width="1214" alt="Image" src="https://github.com/user-attachments/assets/7739ac51-ffe8-4709-8a39-f38efea147c1" />




#!/bin/bash
ntrain=5
save_dir="mmlu_eval_results/"
global_record_file="mmlu_eval_results/eval_record_collection.csv"
#model="/home/test/.cache/modelscope/hub/deepseek-ai/DeepSeek-V2-Lite"
# model="/home/data/Liam/qwen"
model="/home/data/Liam/deepseek"
peft_model="."
prune_result="."
#prune_result="/home/test/test/ghc/moe_prune/MoE_unsupervised_pruning/pruned_result/DeepSeek-V2-Lite/sample_1000/cluster_12/hierarchical_prune/rate_0.5"
#prune_result="/home/test/test/ghc/moe_prune/MoE_unsupervised_pruning/pruned_result/DeepSeek-V2-Lite/sample_1000/cluster_12/hierarchical_prune/rate_0.1/cluster_12/hierarchical_prune/rate_0.1"
#prune_result="/home/test/test/ghc/moe_prune/MoE_unsupervised_pruning/pruned_result/DeepSeek-V2-Lite/sample_1000/no_layerwise_prune/cluster_12/hierarchical_prune/rate_0.2"
#prune_result="/home/test/test/ghc/moe_prune/MoE_unsupervised_pruning/pruned_result/DeepSeek-V2-Lite/sample_1000/seer_prune/rate_0.2/c4_prune.json"
# prune_result="/home/data/Liam/moe/pruned_result/qwen/sample_1000/cluster_12/hierarchical_prune/rate_0.1/cluster_6/hierarchical_prune/rate_0.1/finance_alpaca_prune.json"
# prune_result="/root/autodl-tmp/rate_0.1/finance_alpaca_prune.json"
#prune_result="/home/test/test/ghc/moe_prune/MoE_unsupervised_pruning/pruned_result/Qwen1___5-MoE-A2___7B/sample_1000/cluster_16/hierarchical_prune/rate_0.1/cluster_16/hierarchical_prune/rate_0.1"
#prune_result="/home/test/test/ghc/moe_prune/MoE_unsupervised_pruning/pruned_result/Qwen1___5-MoE-A2___7B/sample_32/hsic_prune/rate_0.2/c4_prune.json"
#prune_result="/home/test/test/ghc/moe_prune/MoE_unsupervised_pruning/pruned_result/Qwen1___5-MoE-A2___7B/sample_1000/seer_prune/rate_0.2/c4_prune.json"
selected_dataset="mmlupro"
selected_subjects="math,computer_science,business"

# selected_subjects="college_mathematics,college_computer_science,high_school_mathematics,high_school_computer_science"
# selected_subjects="high_school_microeconomics,high_school_macroeconomics,econometrics"
#selected_subjects="college_mathematics,college_computer_science,high_school_mathematics,high_school_computer_science,high_school_microeconomics,high_school_macroeconomics,econometrics"
gpu_util=0.6
batch_size=32
mode="transformers"
CUDA_VISIBLE_DEVICES="0,1"
timestamp=$(date +"%Y%m%d_%H%M%S")
log_file="mmlu_eval_log_$timestamp.out"

CUDA_VISIBLE_DEVICES=$CUDA_VISIBLE_DEVICES nohup python evaluate_from_local.py \
     --ntrain $ntrain \
     --selected_subjects $selected_subjects \
     --selected_dataset $selected_dataset \
     --save_dir $save_dir \
     --model $model \
     --peft_model $peft_model \
     --prune_result $prune_result \
     --global_record_file $global_record_file \
     --gpu_util $gpu_util \
     --batch_size $batch_size \
     --cuda_visible_device $CUDA_VISIBLE_DEVICES \
     --mode $mode > "$log_file" 2>&1 &

pid=$!
echo "Started evaluation script with nohup (PID=$pid)"
echo "Logging to: $log_file"
echo "------------------------------"
echo "Press Ctrl+C to stop viewing logs."
echo "------------------------------"

tail -f "$log_file"

**Comment by Fighoture** [comment]
The result you circled corresponds to the performance without any pruning techniques. However, based on your configuration, you are running the pruned version of the code, so you should refer to the **last row** in each of the two tables:

0.4515 --- 0.4015 in the table
0.5350 --- 0.4816 in the table

Seems like your result is even better than mine. If you want to run models without any pruning, you should assign prune_result as "." for inference.

Additionally, the results are not floating due to the inherent randomness in LLM inference. Therefore, slight fluctuations in performance are expected.

**Comment by Liam-L2** [comment]
Thank you for the clarification regarding the pruning settings and the inherent randomness of LLM inference.

Just to confirm — since there is randomness involved, would it be appropriate for me to rerun your reported configurations (including the unpruned setting) in my own environment to obtain the baseline results, rather than strictly using the results reported in your paper? I want to make sure I'm comparing under consistent conditions.

Please let me know if this approach is acceptable, or if I should directly use the reported results from your paper as the reference instead.

Thanks again for your help!

**Comment by Fighoture** [comment]
Yes, you can reproduce that on your environment and all requirements are the repo I mentioned before. But the above configuration you used is the result for pruned model due to your configuration of pruned_result. You should assign it as "." for unpruned model for further contrast

**Comment by Liam-L2** [comment]
#!/bin/bash
ntrain=5
save_dir="mmlu_eval_results/"
global_record_file="mmlu_eval_results/eval_record_collection.csv"

model="/home/data/Liam/deepseek"
peft_model="."
prune_result="."
selected_dataset="mmlu"

selected_subjects="college_mathematics,high_school_mathematics"
gpu_util=0.6
batch_size=2
mode="transformers"
CUDA_VISIBLE_DEVICES="0,1"
timestamp=$(date +"%Y%m%d_%H%M%S")
log_file="mmlu_eval_log_$timestamp.out"


CUDA_VISIBLE_DEVICES=$CUDA_VISIBLE_DEVICES nohup python evaluate_from_local.py \
     --ntrain $ntrain \
     --selected_subjects $selected_subjects \
     --selected_dataset $selected_dataset \
     --save_dir $save_dir \
     --model $model \
     --peft_model $peft_model \
     --prune_result $prune_result \
     --global_record_file $global_record_file \
     --gpu_util $gpu_util \
     --batch_size $batch_size \
     --cuda_visible_device $CUDA_VISIBLE_DEVICES \
     --mode $mode > "$log_file" 2>&1 &

pid=$!
echo "Started evaluation script with nohup (PID=$pid)"
echo "Logging to: $log_file"
echo "------------------------------"
echo "Press Ctrl+C to stop viewing logs."
echo "------------------------------"


tail -f "$log_file"



Thanks for the clarification! I have followed your suggestion and set prune_result as "." to run the unpruned model in my environment. However, I still find that the results I obtained from the unpruned model are not entirely consistent with the ones reported in the table of your paper. I’m wondering if this might be due to some differences in the environment setup or other factors?

Also, I’d like to ask about the two evaluation scripts: eval_deepseekv2lite_mmlu_batch.sh and eval_deepseekv2lite_mmlupro.sh. I didn’t find corresponding results for these in the paper. When I ran experiments using these scripts, the performance seemed worse than the basic MMLU evaluation. Are these results included in the paper or were they used for additional/internal analysis?

As I’m still a beginner in this area, I really appreciate your detailed guidance and your time in helping me understand these issues better. Thank you again!

<img width="384" alt="Image" src="https://github.com/user-attachments/assets/11644370-62ce-4c92-8c54-7e91d3d59a80" />

**Comment by Fighoture** [comment]
About the result, you should make sure the pruning configuration is right too, not just benchmark part. If there is a big gap between your result and the paper mentioned, pls give me the concrete experiment, including pruning method, subject name and corresponding configuration, etc. Then I can help you solve the issue.

About eval_deepseekv2lite_mmlupro.sh. This is a script for another dataset but we did not test comprehensively. Base on part of the result, it is normal that this dataset is worse because mmlupro is a harder task than mmlu.

**Comment by Liam-L2** [comment]
<img width="1083" alt="Image" src="https://github.com/user-attachments/assets/0662a71d-1754-4bb4-bbcb-44aaf5c4afa8" />

Thank you for your help so far. I want to confirm one detail to ensure I am reproducing the results correctly.

Are the baseline results in your paper based on exactly these two models from Huggingface:

https://huggingface.co/deepseek-ai/DeepSeek-V2-Lite
https://huggingface.co/Qwen/Qwen1.5-MoE-A2.7B


Or did you use any other versions, such as fine-tuned or chat versions of these models for evaluation?

Clarifying this will help me understand whether I am using the correct checkpoints for reproduction.

Thanks again for your guidance!

**Comment by Fighoture** [comment]
The links for model are accurate

**Comment by Liam-L2** [comment]
I have one more question regarding the environment setup.

To what extent do the versions of dependencies (e.g., torch, transformers, CUDA, etc.) affect the final evaluation results? Do I need to exactly match the environment you used in the paper to reproduce the same results? Or is a similar setup generally sufficient?

I want to make sure that any discrepancies I observe are not due to environment mismatches.

Thank you again for your time and support!

**Comment by Fighoture** [comment]
All the package version has been listed in the readme file, while torch is 2.3.x and cuda is 12.1. I do not think that would cause so much difference. You should ensure the configuration in the pruning part is all right, like layerwise and global, pruning rate, and cluster number, etc.

**Comment by Liam-L2** [comment]
Thanks again for your continued support.

At this point, I’d like to clarify that the main issue I’m facing is not related to evaluating the pruned model. Instead, the main difficulty lies in reproducing the baseline (unpruned) model performance. Specifically, the results I get when evaluating the base model differ quite a bit from the values shown in your paper’s tables, even though I believe I have followed the correct configuration.

This makes it difficult for me to assess the real effect of pruning, since the baseline itself doesn’t align.

Looking forward to your advice, and thank you again!

**Comment by Fighoture** [comment]
I just review the code. You can try compute_accuracy.py to show the accuracy, which has some difference with the logic in the evaluate code. Sorry for forgetting this script because I have not reviewed the code for several months...

python compute_accuracy.py directory_of_your_eval_result

**Comment by Liam-L2** [comment]
Thank you for the reminder! I’ve already tried using the compute_accuracy.py script as you suggested. The result I got is shown in the screenshot below.

However, the final accuracy is still quite different from what is reported in the paper (e.g., 34.03%), and the gap is relatively large. I’ve double-checked the settings and input files, and I’m still trying to figure out what might be causing the inconsistency.

Please let me know if there are any other steps or details I might be overlooking. I really appreciate your continued guidance!

<img width="1222" alt="Image" src="https://github.com/user-attachments/assets/9fc88e1c-cf10-42b7-b7b7-679e8a9d63e9" />

<img width="1077" alt="Image" src="https://github.com/user-attachments/assets/3cabca17-7397-4873-a061-8711998e5944" />

**Comment by Fighoture** [comment]
There might be inconsistent but according to your result, I think it is common that the accuracy fluctuate 5% for this experiments because we only select 2-3 subjects. You can delete the predict result and make the inference again then you can see the result is different. The result in the paper is the average score for 3 different times inferences. And you could pay more efforts on the gap between pruned model and unpruned model in the same experiment setting.

**Comment by Liam-L2** [comment]
<img width="166" alt="Image" src="https://github.com/user-attachments/assets/7ec084ea-fb5c-4cd8-9701-f4ebb1a9358e" />

I also have a quick question regarding the pruning setup.

For the tasks like GSM8K and HumanEval, may I ask which dataset was used during the pruning process? Was it the C4 dataset or some task-specific data?

Thanks again for your clarification!

**Comment by Liam-L2** [comment]
<img width="332" height="558" alt="Image" src="https://github.com/user-attachments/assets/53e997ba-ac3d-4d1f-9e7e-9c16692bc94a" />

I also have a quick question regarding the pruning setup.

For the tasks like GSM8K and HumanEval, may I ask which dataset was used during the pruning process? Was it the C4 dataset or some task-specific data?

Thanks again for your clarification!

**Comment by Fighoture** [comment]
During pruning, we used MathInstruct and code_alpaca_20k as input data and gather the intermediate embedding for pruning desition
